### PR TITLE
Add JS String data

### PR DESCRIPTION
**Pull request by Elchi3**
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String and 51 sub pages.

### CODE REVIEW

File: javascript/builtins/String.json [hunk 1]
```diff
@@ -0,0 +1,3314 @@
...
+                  }
+                },
+                "status": {
+                  "experimental": false,
+                  "standard_track": true,
+                  "obsolete": false
+                }
+              },
+              "unicode_point_escapes": {
+                "desc": "Unicode point escapes \\u{xxxxxx}",
```
Comment (wbamberg) [inline javascript/builtins/String.json]: "codepoint" not "point"?

File: javascript/builtins/String.json [hunk 2]
```diff
@@ -0,0 +1,3314 @@
...
+                  "experimental": false,
+                  "standard_track": false,
+                  "obsolete": true
+                }
+              }
+            }
+          },
+          "search": {
+            "__compat": {
+              "basic_su
```
Comment (wbamberg) [inline javascript/builtins/String.json]: this is called "locale" in the wiki, not "locales"

Comment (wbamberg) [inline javascript/builtins/String.json]: same here

**Comment by Elchi3** [comment]
As always, excellent catches. I fixed those now. Thanks @wbamberg!

**Pull request merged**